src/matcha_ml/storage/azure_storage.py

```diff
@@ -195,3 +195,17 @@ def _sync_remote(self, container_name: str, blob_set: Set[str]) -> None:
         # Remove blobs that are not present in the local `src_folder_path``
         for blob in blob_set:
             container_client.delete_blob(blob)
+
+    def _sync_local(self, dest_folder_path: str) -> None:
+        """Synchronizes the local .matcha folder with the remote storage files.
+
+        Args:
+            dest_folder_path (str): Path to folder containing matcha resources
+        """
+        # Clears the local matcha directory by removing all files,
+        # ensuring that it exclusively contains the files retrieved from Azure remote storage
+        if os.path.exists(dest_folder_path):
+            matcha_template_dir = os.path.join(os.getcwd(), ".matcha")
+            shutil.rmtree(matcha_template_dir)
+        else:
+            print("Skipping deleting the '.matcha' folder")
```

Comment:
We can avoid priniting, it does not provide any useful information.

Reply:
We should not raise the error, as we observed this while testing on a different user. Since a `.matcha` folder might not be present on different user, so workflow should not delete `.matcha` (nor raise any error) but simply ignore if `.matcha/infrastructure/resources` folder does not exist.